purchases/src/main/kotlin/com/revenuecat/purchases/Purchase.kt

```diff
@@ -0,0 +1,77 @@
+package com.revenuecat.purchases
+
+import android.app.Activity
+import com.revenuecat.purchases.interfaces.ProductChangeCallback
+import com.revenuecat.purchases.models.GoogleProrationMode
+import com.revenuecat.purchases.models.PurchasingData
+import com.revenuecat.purchases.models.StoreProduct
+import com.revenuecat.purchases.models.SubscriptionOption
+
+open class Purchase(builder: Builder) {
```

Comment:
i don't think we should name it this, too confusing with Purchases. will have to think of a better name

Reply:
`PurchaseParams`?

Reply:
Also, I would generate a `toString`, `equals` and `hashCode`

Reply:
i like that, done!

and just curious - is there a reason the PurchasesConfiguration doesn't need those methods?

Reply:
actually, i don't think there'd be any custom equality logic here.. unless i'm missing something..